evalml/tests/automl_tests/test_automl.py

```diff
@@ -1805,7 +1804,8 @@ def test_pipelines_in_batch_return_nan(
         X_train=X,
         y_train=y,
         problem_type="binary",
-        max_batches=3,
+        max_batches=2,
+        automl_algorithm="iterative",
```

Comment:
Why was this change required?

Reply:
It was set to run the default algorithm even though we had the iterative one mocked out! I updated it so that we're calling the mocks now.